specification.md

```diff
@@ -83,48 +87,64 @@ A Bark tree can consist of four elements:
++ Strictly-named [YAML](https://en.wikipedia.org/wiki/YAML) plain text files, containing metadata following a specific structure and naming format
 
 Standard filesystem directories support hierarchical organization of
-datasets and plaintext YAML files provide metadata attributes. BARK specifies the layout used to store data
-within this framework, while allowing the user to add metadata specific to an
-application.
+datasets, and plaintext YAML files provide metadata attributes. BARK specifies the layout used to store data
+within this framework, plus a minimal set of metadata necessary to make sense of the datasets,
+while allowing the user to add metadata specific to an application.
 
 
 ### Roots
 
-A *root* is a top-level directory containing a `meta` file and entries.
-There are no required attributes in the root `meta` file. Root directories must not
-contain datasets.
+A *root* is a top-level directory containing a `meta` file and zero or more
+*entries*.
+
+Root directories must not contain datasets.
+
+#### Root metadata
+
+There are no required attributes in the root `meta` file.
+
+Any attributes the user finds helpful may be included.
 
 ### Entries
 
-An *entry* is defined as an abstract grouping of zero or more *datasets* that
-all share a common start time. Each *entry* shall be represented by a directory.
+An *entry* is an abstract grouping of zero or more *datasets* that
+all share a common start time. Each entry shall be represented by a directory.
 The directory shall contain all the data objects associated with that entry, 
-and all the metadata associated with the entry, stored
-as YAML key-value pairs in a file named `meta`. The following attributes are required:
+and all the metadata associated with the entry.
+
+Entry directories must not contain root or entry directories.
+Any subdirectories are ignored.
+
+#### Entry metadata
+
+An entry's metadata must be stored in a file named `meta`.
+
+The following attributes are **required**:
 
 -   **timestamp:** The start time of the entry. This attribute shall consist of a
-    two-element array with the first element indicating the number of
-    seconds since January 1, 1970 UTC, and the second element
+    two-element array, with the first element indicating the POSIX time (number of
+    seconds since January 1, 1970 UTC), and the second element
     indicating the rest of the elapsed time, in microseconds. Must
     have at least 64-bit integer precision.
--   **uuid:** A universally unique ID for the entry (see [RFC 4122](http://tools.ietf.org/html/rfc4122.html)). Must be stored
-    as a 128-bit integer or a 36-byte `H5T_STRING` with `CTYPE` of
-    `H5T_C_S1`. The latter is preferred, as 128-bit integers are not
-    supported on many platforms.
+-   **uuid:** A universally unique ID for the entry (see [RFC 4122](http://tools.ietf.org/html/rfc4122.html)).
+    Must be stored as a string in the `meta` file; internal programmatic
```

Comment:
I feel like this description could be better, see https://en.wikipedia.org/wiki/Universally_unique_identifier#Format